### ISSUE TITLE
Templates documentation

### ISSUE THREAD

**Issue by saimn** [Samreay/Marz#134]
There are a few field in the templates definition that are not described, and seems unused: features, featureWeights ?

Some other are used but have no description: z_end2, weights, isStar, isQuasar

**Comment by Samreay** [comment]
I've added a line or two to the doco, but to go through it here as well:
- `z_end2` is used in conjunction with `z_end` for stellar spectra. If we simply used `z_end` for stars, because their allowed redshift range is so small, the RMS normalisation of their xcor array would give them massive peaks. Instead, we normalise over a larger redshift range, but restrict matched redshifts to a separate smaller range.
- `weights` is used to adjust the xcor peak height when given prior information about the object.
- `isStar` is used only for QOP purposes, as the flag system used by OzDES has (for example) a 4 for a confident redshift of a galactic object, but a 6 for a confident redshift of a stellar object. `isStar` simply determines which number to give.
- `quasar`, unlike `isStar` actually changes the fitting methodology used so that broad features are not subtracted.

I have also commented out two other properties I never got to work: `features` and `featureWeights`. The plan for these was to be able to construct artificial spectra using this information, so I could test that there was no bias in fitting to the the templates. However, issues (such as outflow for quasars) make this physical impractical, and I instead test systematic offsets using completely artificial spectra.
